pkg/controller/datavolume-controller.go

```diff
@@ -2558,10 +2613,22 @@ func getDefaultAccessModes(c client.Client, storageClass *storagev1.StorageClass
 		return nil, errors.Wrap(err, "no accessMode defined on DV, cannot get StorageProfile")
 	}
 
-	if len(storageProfile.Status.ClaimPropertySets) > 0 &&
-		len(storageProfile.Status.ClaimPropertySets[0].AccessModes) > 0 {
-		accessModes := storageProfile.Status.ClaimPropertySets[0].AccessModes
-		return accessModes, nil
+	if len(storageProfile.Status.ClaimPropertySets) > 0 {
+		// check for access modes matching with given pvc volume mode
+		defaultAccessModes := []corev1.PersistentVolumeAccessMode{}
+		for _, cps := range storageProfile.Status.ClaimPropertySets {
+			if cps.VolumeMode != nil && *cps.VolumeMode == *pvcVolumeMode {
```

Comment:
can pvcVolumeMode be nil? If so then this is a npe

Reply:
we call this function after making sure: (pvcSpec.VolumeMode == nil || *pvcSpec.VolumeMode == "") is false and pvcVolumeMode == pvcSpec.VolumeMode

Reply:
but I can add a check to be on the safe side is someone will use it in the future wrong..

Reply:
Yeah that is exactly what I was thinking never assume that someone else will not call your function from somewhere that doesn't do the check. Just check in the function to be sure.